lib/common/xml.c

```diff
@@ -848,8 +849,17 @@ string2xml(const char *input)
 
     xmlCtxtResetLastError(ctxt);
     xmlSetGenericErrorFunc(ctxt, pcmk__log_xmllib_err);
+
+    xmlSetGenericErrorFunc(ctxt, NULL);
     output = xmlCtxtReadDoc(ctxt, (pcmkXmlStr) input, NULL, NULL,
-                            PCMK__XML_PARSE_OPTS);
+                            PCMK__XML_PARSE_OPTS_WITHOUT_RECOVER);
+    xmlSetGenericErrorFunc(ctxt, pcmk__log_xmllib_err);
```

Comment:
It'll be fine to let the library log the errors, so the user can see exactly what they've got wrong. If it succeeds with RECOVER, we can log a warning that we successfully recovered from the XML errors, but a future release will treat this as a fatal failure.